resources/assets/js/components/ApplicantProfile/Skills/List.tsx

```diff
@@ -0,0 +1,212 @@
+      skillCategories
+        .filter((skillCategory) => skillCategory.depth !== 1) // Filter out non-top-level categories.
+        .reduce(
+          (
+            accumulator: {
+              [catId: number]: {
+                title: localizedFieldNonNull | undefined;
+                skills: Skill[];
+              };
+            },
+            currentCategory,
+          ) => {
+            if (currentCategory.parent_id !== null) {
+              // Ensure there is a parent category set.
+              const titleValue = skillCategories
+                .filter((skillCategory) => skillCategory.depth === 1)
+                .find((x) => x.id === currentCategory.parent_id)?.name;
+
+              const skillsValue = skills.filter(
+                (currentSkill) =>
+                  currentSkill.id ===
+                  skillSkillCategories.find(
+                    (y) =>
+                      y.skill_id === currentSkill.id &&
+                      y.skill_category_id === currentCategory.id,
+                  )?.skill_id,
+              );
+
+              if (hasKey(accumulator, currentCategory.parent_id)) {
+                accumulator[currentCategory.parent_id].skills = [
+                  ...skillsValue,
+                  ...accumulator[currentCategory.parent_id].skills,
+                ]; // Add child category to existing parent category group.
+              } else {
+                accumulator[currentCategory.parent_id] = {
+                  title: titleValue,
+                  skills: skillsValue,
+                }; // Add child category to new parent category group.
+              }
+            }
+            return accumulator;
+          },
+          [],
+        ),
+    [skillSkillCategories, skillCategories, skills],
+  );
+
+  const skillCategoriesGrouped = Object.keys(skillCategoriesGroupedObject).map(
+    (z) => {
+      if (
+        skillCategoriesGroupedObject[z] !== null &&
+        (!skillCategoriesGroupedObject[z].skills ||
+          skillCategoriesGroupedObject[z].skills.length > 0)
+      ) {
+        return skillCategoriesGroupedObject[z]; // Return only non-empty objects and groups that have skills.
+      }
+      return false;
+    },
+  );
+
+  const deleteSkill = (): Promise<void> => {
+    return Promise.resolve();
+  };
+
+  return (
+    <div>
+      <p data-h2-margin="b(bottom, .5)">
+        {intl.formatMessage(messages.sortBy)}{" "}
+        <SortButton
+          text={intl.formatMessage(messages[SortTypes.Group])}
+          active={sortType === SortTypes.Group}
+          handleClick={() => setSortType(SortTypes.Group)}
+        />{" "}
+        <SortButton
+          text={intl.formatMessage(messages[SortTypes.Alpha])}
+          active={sortType === SortTypes.Alpha}
+          handleClick={() => setSortType(SortTypes.Alpha)}
+        />
+      </p>
+      {sortType === "group" &&
+        skillCategoriesGrouped.map((category) => (
+          <div key={category.catId}>
```

Comment:
`catId` is actually undefined. It was never a property attached to elements, it was the key of that element (or index, when its an array). You could get the index like so:
```
skillCategoriesGrouped.map((category, i) => (
      <div key={i}>
```
But I don't think that is the best key to use.